batcher/aligned-sdk/src/communication/batch.rs

```diff
@@ -0,0 +1,143 @@
+use std::{collections::HashSet, time::Duration};
+
+use futures_util::StreamExt;
+use log::{debug, error};
+use tokio::time::timeout;
+
+use crate::{
+    core::errors,
+    core::types::{
+        AlignedVerificationData, BatchInclusionData, VerificationCommitmentBatch,
+        VerificationDataCommitment,
+    },
+    eth::BatchVerifiedEventStream,
+};
+
+const AWAIT_BATCH_VERIFICATION_TIMEOUT: u64 = 60;
+
+pub async fn handle_batch_inclusion_data<'s>(
+    batch_inclusion_data: BatchInclusionData,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+    stream: &mut BatchVerifiedEventStream<'s>,
+    verified_batch_merkle_roots: &mut HashSet<Vec<u8>>,
+) -> Result<(), errors::SubmitError> {
+    debug!("Received response from batcher");
+    debug!(
+        "Batch merkle root: {}",
+        hex::encode(batch_inclusion_data.batch_merkle_root)
+    );
+    debug!("Index in batch: {}", batch_inclusion_data.index_in_batch);
+
+    let verification_data_commitment = verification_data_commitments_rev.pop().unwrap_or_default();
+
+    if verify_response(&verification_data_commitment, &batch_inclusion_data) {
+        aligned_verification_data.push(AlignedVerificationData::new(
+            &verification_data_commitment,
+            &batch_inclusion_data,
+        ));
+    }
```

Comment:
change to a call to `handle_batch_inclusion_data_without_await`

Reply:
Solved in [f88fc70](https://github.com/yetanotherco/aligned_layer/pull/583/commits/f88fc708e662373cd9a8a2895717618bca695119).